File: frontend/src/types/rag.ts
export interface RAGCitation {
  documentId: string;
  documentName: string;
  documentType: string;
  pageNumber: number;
  clauseNumber?: string | null;
  clauseTitle?: string | null;
  excerpt: string;
  relevanceScore: number;
}

export interface RAGQueryRequest {
  query: string;
  topK?: number;
}

export interface RAGQueryResponse {
  query: string;
  answer: string;
  grounded: boolean;
  status: "GROUNDED" | "INSUFFICIENT_EVIDENCE";
  confidence: number;
  citations: RAGCitation[];
  bundleId: string;
  disclaimer: string;
}

export interface ChunkItem {
  id: string;
  bundleId: string;
  documentId: string;
  pageNumber: number;
  clauseNumber?: string | null;
  clauseTitle?: string | null;
  chunkType: string;
  chunkText: string;
  hasEmbedding: boolean;
  createdAt: string;
}

File: backend/app/schemas/rag.py
from typing import List, Optional
from pydantic import BaseModel, ConfigDict, Field


class EvidenceCitationSchema(BaseModel):
    documentId: str
    documentName: str
    documentType: str
    pageNumber: int
    clauseNumber: Optional[str] = None
    clauseTitle: Optional[str] = None
    excerpt: str
    relevanceScore: float

    model_config = ConfigDict(from_attributes=True)


class RAGQueryRequestSchema(BaseModel):
    query: str = Field(..., min_length=2, description="Natural language question regarding the transaction documents")
    topK: Optional[int] = Field(5, ge=1, le=20, description="Number of context chunks to retrieve")

    model_config = ConfigDict(from_attributes=True)


class RAGQueryResponseSchema(BaseModel):
    query: str
    answer: str
    grounded: bool
    status: str  # "GROUNDED", "INSUFFICIENT_EVIDENCE"
    confidence: float
    citations: List[EvidenceCitationSchema] = []
    bundleId: str
    disclaimer: str = "ClauseGuard is an informational verification platform and does not provide legal advice."

    model_config = ConfigDict(from_attributes=True)


class DocumentIndexingStatusSchema(BaseModel):
    bundleId: str
    documentId: str
    documentName: str
    chunksIndexed: int
    message: str

    model_config = ConfigDict(from_attributes=True)


class ChunkResponseSchema(BaseModel):
    id: str
    bundleId: str
    documentId: str
    pageNumber: int
    clauseNumber: Optional[str] = None
    clauseTitle: Optional[str] = None
    chunkType: str
    chunkText: str
    hasEmbedding: bool
    createdAt: str

    model_config = ConfigDict(from_attributes=True)
